addon-entwicklung.md

```diff
@@ -1 +1,169 @@
 # AddOn Entwicklung
```

Comment:
Schreibweise: AddOn

Reply:
erledigt

Reply:
Die Beschreibung der boot.php ist sachlich falsch. boot.php wird nicht erst bei Verwendung des AddOn ausgeführt, sondern bei jedem Start der REDAXO Instanz sobald das entsprechende AddOn aktiviert ist.

Reply:
Ich empfehle einen Einleitungstext dem ganzen voran zu stellen. Hier wird sofort auf Details eingegangen ohne dem User zu sagen welche Idee hinter den AddOn Konzept steht.

Reply:
AddOns werden nicht mit der Installation registriert sondern mit des Aktivierung. Deshalb kann ich ein AddOn auch deaktivieren ohne es auch deinstallieren zu müssen. Eine Aktivierung ist zwar Teil des Installationsprozesses, aber diese kann später im AddOnManager aufgehoben werden.
Grundsätzlich ist der tatsächliche zeitliche Ablauf eine AddOn Einbindung sogar noch anders:
Beim Start einer REDAXO Instanz werden die package.yml Dateien aller aktiven AddOns zum Registrierungsprozess herangezogen.

Reply:
Gehören Auslassungen über MarkitUp wirklich an diese Stelle?

Reply:
Beschreibung zu boot.php ist gefixt.  
Fehlende Abschnitte gibt es noch viele, eine Einleitung ist da sicherlich nicht ausgeschloßen.  
MarkDown / MarkitUp ist eine wesentliche Komponente bei der Entwicklung, daher erlaube ich mir die "Auslassungen" darüber zu verteidigen.

Reply:
Das ist ja auch ein wichtiger Bestandteil einer Dokumentation, da gebe ich dir völlig recht. An dieser Stelle sehe ich das allerdings nicht so wirklich, zumal dieses Tool nicht nur in der Addon Entwicklung benötigt würde.

Reply:
Die Ansicht kann ich unter Berücksichtigung Deines Vorschlags "Best Practice" jetzt nicht unterstützen, allerdings wird das gesamte Thema Entwicklung in Zukunft sicherlich in Unterseiten aufgeteilt. Im momentanen Stand ist die Gewichtung des Markdown-Themas sicher überproportional aber mit dem Hinzufügen weiterer Themen wird diese Gewichtung verschwinden. Ein Verschieben des Abschnitts halte ich für denkbar, aber meiner Ansicht nach dann in ein anderes Kapitel als in "Best Practice".  
Das Yaml-Kapitel wird sich evtl ähnlich, wahrscheinlich sogar umfangreicher entwickeln und logisch könnte man beide in "spezielle Dateiformate" einordnen. Sicherlich kann man MarkDown auch in "Dokumentation erstellen" einordnen, da bin ich offen.  
Zum momentanen Zeitpunkt würde ich, sofern **jetzt** keine überzeugende Alternative existiert, den Abschnitt **erst mal** dort lassen.

Reply:
Da MakItUp ja als AddOn bereits existiert, ist das eventuell eine sinnvolle Ergänzung im dortigen Help oder in einer Doku-Pages Seite.

Reply:
Ich denke MarDown findet unabhängig vom AddOn verwendung, daher ist ein Abschnitt über "Dokumentation erstellen" auf jeden Fall sinnvoll, und zwar losgelöst von spezifischen AddOns.  
Ich gehe sogar soweit, daß AddOns eher auf die allgemeine Doku verweisen sollten als andersherum. Der Spezifische Inhalt der Abschnitte (in diesem Fall meines Abschnittes über MarkDown) ist erst einmal unwichtig, da es um die logische Struktur der Doku geht.

Wenn man textile installiert hat, kann man MarkItUp nicht installieren, somit fällt die Betrachtung möglicher Doku oder Beispiele am konkreten AddOn weg (für den Normalbenutzer), meine obige Forderung hat damit ganz konkrete Gründe, um die Doku möglichst Jedem ohne Barriere zur Verfügung zu stellen.

Reply:
Auch das Erscheinen neuer REDAXO Versionen kann den Fall einiger AddOns bewirken, wenn diese nicht aktualisiert werden. Die Doku wäre dann dem Entwicklungsstand einzelner AddOns ausgeliefert, was sicherlich nicht erstrebenswert ist.

Reply:
Wenn die Doku über mehrere Quellen hinweg verteilt ist, erschwert das die Lesbarkeit unnötig.
Sogar mein Beispiel mit config.aml sollte den entsprechenden Kode direkt in der Doku zeigen und nicht erfordern, daß man das AddOn runterladen muss.
Es wäre schön, wenn man in MarkDown eine Codebox integrieren könnte, damit der Kode die Seite nicht quasi-unlesbar macht, ansonsten muss man den Kode immer limitieren.

Reply:
Worüber sprechen wir jetzt eigentlich? Ich dachte es geht um die RADAXO Doku und nicht über eine allgemeine Anleitung für das Entwickeln von CMS Frameworks

Reply:
@gupi lass mich wissen, wenn Du was Konkretes hast. Zum gegenwärtigen Zeitpunkt denke ich dass  alle strittigen Punkte geklärt sind. Das "das [allgemeine] Entwickeln von CMS Frameworks" steht hier nicht zur Debatte.